pkg/process/checks/enabled_checks_linux_test.go

```diff
@@ -37,3 +38,41 @@ func TestProcessEventsCheckEnabled(t *testing.T) {
 		assertNotContainsCheck(t, enabledChecks, ProcessEventsCheckName)
 	})
 }
+
+func TestConnectionsCheckLinux(t *testing.T) {
+	// Make sure the connections check is disabled on the core agent
+	// and enabled in the process agent when process checks run in core agent
+	t.Run("run in core agent", func(t *testing.T) {
+		cfg, scfg := config.Mock(t), config.MockSystemProbe(t)
+		cfg.SetWithoutSource("process_config.process_collection.enabled", true)
+		cfg.SetWithoutSource("process_config.run_in_core_agent.enabled", true)
+		scfg.SetWithoutSource("network_config.enabled", true)
+		scfg.SetWithoutSource("system_probe_config.enabled", true)
+
+		flavor.SetFlavor("agent")
```

Comment:
Flavor is a global setting, so just to be safe you should reset the flavor to whatever it was set to previously, with a defer/cleanup.

Reply:
Done